Handler/DownloadHandler.php

```diff
@@ -40,12 +41,21 @@ public function downloadObject($object, string $field, ?string $className = null
             $fileName = $mapping->readProperty($object, 'originalName');
         }
 
-        $file = $mapping->getFile($object);
+        $mimeType = $mapping->readProperty($object, 'mimeType');
+
+        if (!$mimeType) {
+            try {
+                $file = $mapping->getFile($object);
+                $file ? $mimeType = $file->getMimeType() : null;
```

Comment:
Same here

Reply:
@garak please elaborate, I do not know what you mean

Reply:
I mean same as noted above: avoid implicit casting

Reply:
@garak done